backend/logic/discount/index.js

```diff
@@ -11,27 +11,32 @@ const { Sequelize, Discount, Order } = require('../../models')
  * @returns {{error: string}|{discount: models.Discount}}
  */
 const validateDiscount = async (code, shop) => {
-  const discounts = await Discount.findAll({
-    where: {
-      [Sequelize.Op.and]: [
-        { status: 'active' },
-        Sequelize.where(
-          Sequelize.fn('lower', Sequelize.col('code')),
-          Sequelize.fn('lower', code)
-        )
-      ],
-      shopId: shop.id,
-      startTime: {
-        [Sequelize.Op.lte]: Date.now()
-      },
-      endTime: {
-        [Sequelize.Op.or]: [
-          { [Sequelize.Op.gt]: Date.now() },
-          { [Sequelize.Op.eq]: null }
-        ]
-      },
-      discountType: !code ? DiscountTypeEnums.payment : undefined
+  const where = {
+    [Sequelize.Op.and]: [
+      { status: 'active' },
+      Sequelize.where(
+        Sequelize.fn('lower', Sequelize.col('code')),
+        Sequelize.fn('lower', code)
+      )
+    ],
+    shopId: shop.id,
+    startTime: {
+      [Sequelize.Op.lte]: Date.now()
+    },
+    endTime: {
+      [Sequelize.Op.or]: [
+        { [Sequelize.Op.gt]: Date.now() },
+        { [Sequelize.Op.eq]: null }
+      ]
     }
+  }
+
+  if (!code) {
```

Comment:
Why would the validateDiscount method ever be called without a code?
It's not obvious to me. Can you add a comment in the code explaining that so that we don't scratch our head in the future when reading this code?

Reply:
Payment-specific discounts don't have a code. Will add a comment.